includes/Admin/Admin_Page.php

```diff
@@ -15,28 +15,88 @@
 class Admin_Page {
 
 	/**
-	 * Initializes hooks.
+	 * Admin AJAX class instance.
+	 *
+	 * @since n.e.x.t
+	 * @var Admin_AJAX
+	 */
+	protected $admin_ajax;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function __construct() {
+		$this->admin_ajax = new Admin_AJAX();
+	}
+
+	/**
+	 * Registers WordPress hooks for the admin page.
 	 *
 	 * @since n.e.x.t
 	 */
 	public function add_hooks() {
 		add_action( 'admin_menu', array( $this, 'add_page' ) );
 		add_filter( 'plugin_action_links', array( $this, 'filter_plugin_action_links' ), 10, 2 );
+
+		$this->admin_ajax->add_hooks();
 	}
 
 	/**
 	 * Registers the admin page under the tools menu.
 	 *
 	 * @since n.e.x.t
+	 *
+	 * @return string The hook identifier for the admin page.
 	 */
 	public function add_page() {
-		add_management_page(
+		$hook = add_management_page(
 			__( 'Plugin Check', 'plugin-check' ),
 			__( 'Plugin Check', 'plugin-check' ),
 			'activate_plugins',
 			'plugin-check',
 			array( $this, 'render_page' )
 		);
+
+		add_action( "load-{$hook}", array( $this, 'initialize_page' ) );
+
+		return $hook;
+	}
+
+	/**
+	 * Initializes page hooks.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function initialize_page() {
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+	}
+
+	/**
+	 * Loads the check's script.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function enqueue_scripts() {
+		wp_enqueue_script(
+			'plugin-check-admin',
+			WP_PLUGIN_CHECK_PLUGIN_DIR_URL . 'assets/js/plugin-check-admin.js',
+			array(),
+			WP_PLUGIN_CHECK_VERSION,
+			true
+		);
+
+		wp_add_inline_script(
+			'plugin-check-admin',
+			'const PLUGIN_CHECK = ' . json_encode(
+				array(
+					'ajaxUrl' => admin_url( 'admin-ajax.php' ),
```

Comment:
Nitpick, but I think sending this value is unnecessary since `ajaxurl` is already set as a global in the admin.